src/Sustainsys.Saml2/Serialization/SamlXmlReader.Attribute.cs

```diff
@@ -40,8 +40,11 @@ protected virtual void ReadElements(XmlTraverser source, SamlAttribute attribute
         while(source.MoveNext(true) 
             && source.EnsureName(Elements.AttributeValue, Namespaces.SamlUri))
         {
-            // TODO: Test + support for null values.
-            attribute.Values.Add(source.GetTextContents());
+            var attributeValue = source.GetTextContents();
+            if (!string.IsNullOrEmpty(attributeValue))
+            {
+                attribute.Values.Add(attributeValue);
+            }
```

Comment:
I am wondering if I need to add a check outside of the while loop to see if Values is populated. If it is not, then we should return null instead of a SamlAttribute object with a populated "Name" but an empty "Values".